llvm/docs/DTLTO.rst

```diff
@@ -60,165 +80,126 @@ Distributors are programs responsible for:
-elements are converted to strings and concatenated. Integers are converted by
-indexing into the specified array.
+To generate the backend compilation commands, the job-specific and common
+arguments are combined.
 
-The example above generates the following backend compilation command for
-`main.o`:
+The above example generates the following backend compilation commands:
 
 .. code-block:: console
 
-    /usr/local/clang -O3 -fprofile-sample-use=my.profdata \
-        -o dtlto.1.51232.native.o -c -x ir dtlto.o \
-        -fthinlto-index=dtlto.1.51232.native.o.thinlto.bc --target=x86_64-sie-ps5
+    clang.exe -c -flto -fprofile-sample-use=my.prof t1.bc \
+        -fthinlto-index=t1.1.7944.native.o.thinlto.bc -o t1.1.7944.native.o \
+        -fproc-stat-report=t1.1.7944.stats.txt
 
-This expansion scheme allows the remote compiler to be changed without updating
-the distributors. For example, if the "args" field in the above example was
-replaced with:
+    clang.exe -c -flto -fprofile-sample-use=my.prof t2.bc \
+        -fthinlto-index=t2.2.7944.native.o.thinlto.bc -o t2.2.7944.native.o \
+        -fproc-stat-report=t2.2.7944.stats.txt
 
-.. code-block:: json
+Temporary Files
+---------------
 
-    "args": [
-        "custom-compiler",
-        "-opt-level=2",
-        "-profile-instrument-use-path=my.profdata",
-        "-output", ["primary_output", 0],
-        "-input", ["primary_input", 0],
-        "-thinlto-index", ["summary_index", 0],
-        "-triple", "x86_64-sie-ps5"
-    ]
+During its operation, DTLTO generates temporary files. Temporary files are
+created in the same directory as the linker's output file:
 
-Then distributors can expand the command line without needing to be updated:
+- **JSON Job Description File**:
 
-.. code-block:: console
+  - Format: ``<linker output file basename>.<PID>.dist-file.json``
+  - Example: ``dtlto.77380.dist-file.json`` (for output file ``dtlto.elf``).
+
+- **Object Files From Backend Compilations**:
+
+  - Format: ``<Module identifier basename>.<Task>.<PID>.native.o``
+  - Example: ``my.1.77380.native.o`` (for bitcode module ``my.o``).
+
+- **Summary Index Shard Files**:
 
-    custom-compiler -opt-level=2 -profile-instrument-use-path=my.profdata \
-        -output dtlto.1.51232.native.o -input dtlto.o \
-        -thinlto-index dtlto.1.51232.native.o.thinlto.bc -triple x86_64-sie-ps5
+  - Format: ``<Module identifier basename>.<Task>.<PID>.native.o.thinlto.bc``
+  - Example: ``my.1.77380.native.o.thinlto.bc`` (for bitcode module ``my.o``).
+
+Temporary files are removed by default.
+
+TODOs
+-----
+
+The following features are planned for DTLTO but not yet implemented:
+
+- Support for the ThinLTO in-process cache.
+- Support for platforms other than ELF and COFF.
+- Support for archives with bitcode members.
+- Support for more LTO configurations; only a very limited set of LTO
+  configurations is supported currently, e.g., support for basic block sections
+  is not currently available.
```

Comment:
Actually, any reason why we haven't forwarded on the `MllvmArgs` of `lto::config`? I think this is actually the reason for one of our downstream DTLTO issues.

Reply:
Something to tackle after the first commit.

Reply:
Actually, I think MaskRay was more keen on forwarding the mllvm arguments that were applied to any clang invocation that invoked the linker to do DTLTO.